File: state_machine_frontend/src/models/event.ts
export interface Rule {
  meta_data_key: string;
  value: any;
  operator: string;
  actions: string[];
}

export interface Event {
  id: string;
  event_name: string;
  next_state_name: string;
  rules: Rule[];
  version: number;
}

export interface NewRule {
  rule_type: string;   
  meta_data_key: string;
  value: any;
  operator: string;
  actions: string[];   
}

File: StateMachine/app/models/event.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import List

from app.models.rule import Rule


@dataclass(slots=True)
class Event:
    id: str
    event_name: str = ""
    next_state_name: str = ""
    rules: List[Rule] = field(default_factory=list)
    version: int = 0


